src/bin/sccache-dist/cmdline/parse.rs

```diff
@@ -0,0 +1,340 @@
+use std::{env, ffi::OsString, fmt, net::SocketAddr, path::PathBuf, str::FromStr};
+
+use anyhow::bail;
+use clap::{Arg, ArgGroup, Command as ClapCommand};
+use sccache::{config, dist::ServerId};
+use syslog::Facility;
+
+use crate::cmdline::{AuthSubcommand, Command};
+
+#[derive(Debug)]
+struct TokenLength(usize);
+
+impl TokenLength {
+    fn as_bytes(&self) -> usize {
+        self.0 / 8
+    }
+
+    fn from_bits(bits: &str) -> anyhow::Result<Self> {
+        let bits: usize = bits.parse()?;
+        if bits % 8 != 0 || bits < 64 || bits > 4_096 {
+            bail!("Number of bits must be divisible by 8, greater than 64 and less than 4096")
+        }
+
+        Ok(Self(bits))
+    }
+}
+
+impl fmt::Display for TokenLength {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+#[derive(Clone, Copy)]
+enum LogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+}
+
+impl LogLevel {
+    fn as_str(&self) -> &'static str {
+        match self {
+            Self::Error => "error",
+            Self::Warn => "warn",
+            Self::Info => "info",
+            Self::Debug => "debug",
+            Self::Trace => "trace",
+        }
+    }
+
+    fn values() -> &'static [Self] {
+        &[
+            Self::Error,
+            Self::Warn,
+            Self::Info,
+            Self::Debug,
+            Self::Trace,
+        ]
+    }
+}
+
+impl FromStr for LogLevel {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let variant = match s {
+            "error" => Self::Error,
+            "warn" => Self::Warn,
+            "info" => Self::Info,
+            "debug" => Self::Debug,
+            "trace" => Self::Trace,
+            _ => bail!("Unknown log level: {:?}", s),
+        };
+
+        Ok(variant)
+    }
+}
+
+impl From<LogLevel> for log::LevelFilter {
+    fn from(log_level: LogLevel) -> Self {
+        match log_level {
+            LogLevel::Error => Self::Error,
+            LogLevel::Warn => Self::Warn,
+            LogLevel::Info => Self::Info,
+            LogLevel::Debug => Self::Debug,
+            LogLevel::Trace => Self::Trace,
+        }
+    }
+}
```

Comment:
There is the crate https://crates.io/crates/clap-verbosity-flag that already does this :)

Reply:
TIL, I'll switch over to that!

Reply:
Took a look at `clap-verbosity-flag` and noticed that it is for the more traditional `-q` and `-v*` style of verbosity flags while `sccache-dist` currently takes `--syslog <LEVEL>` instead. This means that switching to `clap-verbosity-flag` would be a breaking change and as v0.3 was released under a month ago, I'm guessing that another breaking change isn't intended quite this soon (and could merit looking into potentially revamping `sccache`'s CLI structure as well)

Reply:
You're probably right one that, I am tempted to defer that change to another PR, but still strongly in favor of making it. I am not aware of other programs using that naming.